### ISSUE TITLE
Linux Support

### ISSUE THREAD

**Issue by cschwem2er** [jrfiedler/stata-kernel#3]
Is Linux Support possible at all? I did not find any Stata automation possibilities for Linux.

**Comment by TiesdeKok** [comment]
To my knowledge Stata Automation is Windows only (the interaction protocol used is developed by Microsoft). A limited work-around would be to use the batch mode but that would not allow persistence across cells unless you were to rebuild a session each time a cell is executed. This obviously brings a lot of overhead so I do not know if it would be feasible.

**Comment by jrfiedler** [comment]
Yeah, as far as I know there's no automation support for Linux. My guess is that a lot of the same functionality can be achieved using other means. There are two methods, though, that might be difficult to imitate or work around: `UtilSetStataBreak`, used to interrupt Stata, and `UtilIsStataFree`, used to determine if Stata has stopped writing output.

**Comment by kylebarron** [comment]
Thanks to @TiesdeKok's suggestion of checking out `pexpect`, I made a [Jupyter kernel for Stata](https://github.com/kylebarron/stata_kernel) that should work on Mac and Linux (and maybe even Windows, I haven't checked out the Stata console on Windows). It doesn't use Stata Automation at all.

I'm guessing it's still worth it to add support for Automation for Windows and Mac. That should be trivial.

<img width="674" alt="image" src="https://user-images.githubusercontent.com/15164633/43681115-c1dc52ce-9818-11e8-9154-49feb6819d7e.png">

**Comment by kylebarron** [comment]
It seems to work with just a few bugs for me on both Mac and Linux!
